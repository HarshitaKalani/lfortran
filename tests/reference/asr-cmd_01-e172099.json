{
    "basename": "asr-cmd_01-e172099",
    "cmd": "lfortran --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/cmd_01.f90",
    "infile_hash": "11c7006f168f19f99bdf735051e716724b244773d6aff7c50eb77a7f",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-cmd_01-e172099.stdout",
    "stdout_hash": "7cc756c578f45107d26eef4f44ff1e1d2cc3a2b0a0c40d293100b50c",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}