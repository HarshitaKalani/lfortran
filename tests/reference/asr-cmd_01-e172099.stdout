(TranslationUnit
    (SymbolTable
        1
        {
            lfortran_intrinsic_builtin:
                (IntrinsicModule lfortran_intrinsic_builtin),
            test_get_command_argument:
                (Program
                    (SymbolTable
                        2
                        {
                            arg:
                                (Variable
                                    2
                                    arg
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Character 1 32 ())
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            command_argument_count:
                                (ExternalSymbol
                                    2
                                    command_argument_count
                                    4 command_argument_count
                                    lfortran_intrinsic_builtin
                                    []
                                    command_argument_count
                                    Private
                                ),
                            count:
                                (Variable
                                    2
                                    count
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Integer 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            get_command_argument:
                                (ExternalSymbol
                                    2
                                    get_command_argument
                                    4 get_command_argument
                                    lfortran_intrinsic_builtin
                                    []
                                    get_command_argument
                                    Private
                                ),
                            i:
                                (Variable
                                    2
                                    i
                                    []
                                    Local
                                    (IntegerConstant 0 (Integer 4))
                                    (IntegerConstant 0 (Integer 4))
                                    Save
                                    (Integer 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                )
                        })
                    test_get_command_argument
                    []
                    [(Assignment
                        (Var 2 count)
                        (FunctionCall
                            2 command_argument_count
                            ()
                            []
                            (Integer 4)
                            ()
                            ()
                        )
                        ()
                    )
                    (Print
                        [(StringConstant
                            "count: "
                            (Character 1 7 ())
                        )
                        (Var 2 count)]
                        ()
                        ()
                    )
                    (SubroutineCall
                        2 get_command_argument
                        ()
                        [((Var 2 i))
                        ((Var 2 arg))
                        (())
                        (())]
                        ()
                    )
                    (Print
                        [(IntrinsicElementalFunction
                            StringTrim
                            [(Var 2 arg)]
                            0
                            (Character 1 32 ())
                            ()
                        )]
                        ()
                        ()
                    )]
                )
        })
    []
)
